cmake/public/utils.cmake

```
@@ -407,7 +407,11 @@ endmacro()
 # Usage:
 #   torch_compile_options(lib_name)
 function(torch_compile_options libname)
-  set_property(TARGET ${libname} PROPERTY CXX_STANDARD 17)
+  if(DEFINED CMAKE_CXX_STANDARD)
+    set_property(TARGET ${libname} PROPERTY CXX_STANDARD ${CMAKE_CXX_STANDARD})
+  else()
+    set_property(TARGET ${libname} PROPERTY CXX_STANDARD 17)
+  endif()
```

Nit (If I read the [doc](https://cmake.org/cmake/help/latest/variable/CMAKE_CXX_STANDARD.html) correctly, if it is defined, there is no need to explicitly set it)
```suggestion
  if(NOT DEFINED CMAKE_CXX_STANDARD)
    set_property(TARGET ${libname} PROPERTY CXX_STANDARD 17)
  endif()
```

---

Almost, there's a small difference.
If any of those target property was set previously, this line would overwrite that.
So I keep that behavior consistent across both code paths to avoid surprises.